### ISSUE TITLE
HLS with not relative segment URL will causing problem on Exo player2 or not ?

### ISSUE THREAD

**Issue by sitthajaid77** [google/ExoPlayer#10053]
I've 2 cases for serving HLS streaming to Exo player 2 would like to ask.

1. HLS serving with separated playlist and segment URL with different end point
![HLS_01](https://user-images.githubusercontent.com/22445538/157821608-d239c251-5a63-4266-8c26-6c1fbac1a112.png)

2. HLS with playlist and segment with the same endpoint(use relative path followed playlist source) 
![HLS_02](https://user-images.githubusercontent.com/22445538/157821664-b80b31aa-19f5-4e7c-8cf7-9fbb086be7f5.png)

Basically, I found almost OTT service use the scenario 2 for serving HLS. I wonder the scenario 1 will cause Exo player 2 the problems or not, such as stream not smooth or video/audio not sync. The best scenario for serving HLS shall be scenario 2 or not ?
Can anybody help to give me some suggestion please.

Thank you

**Comment by ojw28** [comment]
ExoPlayer will correctly handle both absolute and relative URLs in playlist files, and I would expect that performance would be similar in both cases, provided that the requests are fulfilled just as quickly. Obviously if scenario 1 incurs significantly increased latency when requesting data from the playlist gateway compared to the CDN, then that might affect performance, but whether that's the case is not really an ExoPlayer question.

One benefit of scenario 2 over scenario 1 is that it will probably allow for socket reuse between requests for playlists and segments, but I wouldn't expect this to make a significant difference.

**Comment by sitthajaid77** [comment]
@ojw28 Thank you so much for your comment. At least, ExoPlayer can handle both scenarios whether 1 or 2. If you need to choose, which solution 1 or 2 you will implement ?

**Comment by ojw28** [comment]
We develop a client side player and do not have extensive experience with server-side streaming architectures. So I don't think we're the right people to answer that question.

**Comment by sitthajaid77** [comment]
@ojw28 Thank you for your answer, even please give me one more help. If I going with scenario 2, do you think Exoplayer can reduce the latency compared with scenario 1 or not. Because the client can re-use socket, right ?
